lua/entities/gmod_wire_expression2/core/string.lua

```diff
@@ -502,3 +502,69 @@ e2function string string:matchFirst(string pattern, position)
 		return Ret or ""
 	end
 end
+
+/******************************************************************************/
+local unpack = unpack
+local isnumber = isnumber
+local utf8_len = utf8.len
+
+__e2setcost(1)
+
+--- Returns the UTF-8 string from the given Unicode code-points.
+e2function string toUnicodeChar(...)
+	local args = { ... }
+	local count = #args
+	if count == 0 then return "" end
+	local codepoints = {}
+	for i = 1, count do
+		local value = args[i]
+		if typeids[i] == "n" and isnumber(value) then
+			value = math_floor(value)
+			if 0 <= value and value <= 0x10FFFF then
+				codepoints[#codepoints + 1] = value
+			end
+		end
+	end
+	self.prf = self.prf + #codepoints * 0.001
+	return utf8_char(unpack(codepoints))
+end
+
+--- Returns the UTF-8 string from the given Unicode code-points.
+e2function string toUnicodeChar(array args)
+	local count = #args
+	if count == 0 then return "" end
+	local codepoints = {}
+	for i = 1, count do
+		local value = args[i]
+		if isnumber(value) then
+			value = math_floor(value)
+			if 0 <= value and value <= 0x10FFFF then
+				codepoints[#codepoints + 1] = value
+			end
+		end
+	end
+	self.prf = self.prf + #codepoints * 0.001
```

Comment:
Probably should use count here instead of #codepoints.

Reply:
I'm unable to produce _holes_ in `codepoints` table, at that particular place, so using length operator is faster.